src/SoundList.js

```diff
@@ -3,10 +3,15 @@ import { Link } from "react-router-dom";
 import { playCircleSharp, timeSharp, downloadSharp } from "ionicons/icons";
 import { IonIcon } from "@ionic/react";
 
-function timeConvert(minutes) {
-  const min = Math.floor(Math.abs(minutes));
-  const secs = Math.floor((Math.abs(minutes) * 60) % 60);
-  return `${(min < 10 ? "0" : "") + min}:${secs < 10 ? "0" : ""}${secs}`;
+function format(lengthInSeconds) {
```

Comment:
This algorithm seems more complicated than the previous one. I'm imagining that the solution could be a simple addition to the existing algorithm.

I'm imagining the structure could look like this:
```
const secs = ...
const min = ...
const hrs = ...
```

Also I think `minutes` is a better name for a param since it isn't coupled to the context of our app. In other contexts, "length" in minutes might not make sense. The goal for our utilities/components should be to have the minimal amount of coupling to our app.

Reply:
I also think `format` isn't a good function name. Readers think "format what"? I think `formatTime` is more clear. The variable names `m` and `h` are also an antipattern. Our convention is to never use single letter variable names. `min`, `hrs`, and `secs` are more clear.

Reply:
> I also think `format` isn't a good function name. Readers think "format what"? I think `formatTime` is more clear. The variable names `m` and `h` are also an antipattern. Our convention is to never use single letter variable names. `min`, `hrs`, and `secs` are more clear.

I used `format` to make it more declarative. I think `format(time)` is more readable than `formatTime(time)`. As far as the variable names go, I can definitely make them longer. They were verbose to begin with, but I shortened them because `prettier` wanted them to be on separate lines. I was trying to reduce the LOC to minimize vertical scrolling, since I read that having to scroll vertically, excessively, reduces readability.

> This algorithm seems more complicated than the previous one. I'm imagining that the solution could be a simple addition to the existing algorithm.
> 
> I'm imagining the structure could look like this:
> 
> ```
> const secs = ...
> const min = ...
> const hrs = ...
> ```
> 
> Also I think `minutes` is a better name for a param since it isn't coupled to the context of our app. In other contexts, "length" in minutes might not make sense. The goal for our utilities/components should be to have the minimal amount of coupling to our app.

I was trying to speed up computation time by having the function `format` compute as little as necessary. If the duration is less than a minute long, we don't even need to worry about formatting the minutes or hours portion of the time.

We should leave `seconds` in the parameter name: [according to the Freesound API Documentation](https://freesound.org/docs/api/resources_apiv2.html#response-sound-instance), `track.duration`, which is the only input we give this function, has a unit of seconds. We can change the parameter title to `durationInSeconds` instead of `lengthInSeconds`. It's what I had initially, but I was again trying to cut back on vertical scrolling, since the longer parameter name caused `prettier` to want to split `if (lengthInSeconds < 60) return `00:${addZeroToFrontOf(lengthInSeconds)}`;` across two lines instead of one.

I, for sure, have to make at least one more change, since pages like [the homepage](https://freesound-player-git-fix-sound-length-formatting.amilajack.vercel.app) look awkward with so much space given to the time for its formatting. It looks okay if the `<SoundList />` has a track over an hour long. If all sounds in a `<SoundList />` are under an hour long, like on the linked page, it doesn't look elegant. I have a fix for that.

Reply:
> I used format to make it more declarative. I think format(time) is more readable than formatTime(time)...

Editors and other tools don't provide autocompletion based on variable names. They're always based on function name:

<img width="927" alt="Screen Shot 2020-11-26 at 7 51 06 PM" src="https://user-images.githubusercontent.com/6374832/100409458-bcd38e80-3020-11eb-85a2-fef670babd1d.png">

When a reader is trying to search fns to get a list of fn's available in the current module, they'll see `format`, which won't make much sense to then since they can't see the params.

TL;DR: params should not be used to infer what a function does. Function name + comments take that responsibility.

> I was trying to speed up computation time by having the function...

This is a subtle form of premature optimization. The formatting of time is far from being the bottleneck of our application so sacrificing simplicity for complexity won't help us much.

> They were verbose to begin with, but I shortened them because prettier wanted them to be on separate lines...

I recommend not changing the code to reflect make it easier for prettier. I used to do this but it always bit me later when trying to understand my code later on. Single variable names are always an antipattern. Adding a few lines of code is fine.

See this section of [Idomatic.js](https://github.com/rwaldron/idiomatic.js/), which I highly recommend reading:
<img width="962" alt="Screen Shot 2020-11-26 at 7 49 29 PM" src="https://user-images.githubusercontent.com/6374832/100409366-80079780-3020-11eb-8259-a62569f6bb04.png">

Reply:
> Editors and other tools don't provide autocompletion based on variable names. They're always based on function name:

I didn't realize editors omitted parameter names when looking up function names. Now I see that it does make more sense to add more detail to a function's name, like you said. My incoming commit will do that. Thanks for the tip!

> I recommend not changing the code to reflect make it easier for prettier. I used to do this but it always bit me later when trying to understand my code later on. Single variable names are always an antipattern. Adding a few lines of code is fine.

For sure. My incoming commit will have a lot more detail. I perused the [Idomatic.js](https://github.com/rwaldron/idiomatic.js/) link you sent me a bit last night, and it seems really useful. Thanks for sending me the link!